packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -394,7 +415,7 @@ export class RunningSummarizer implements IDisposable {
     private trySummarize(
         reason: SummarizeReason,
         cancellationToken = this.cancellationToken): void {
-        if (this.summarizingLock !== undefined) {
+        if (this.summarizingLock !== undefined || this.refreshSummaryAckLock !== undefined) {
             // lockedSummaryAction() will retry heuristic-based summary at the end of current attempt
             // if it's still needed
             this.tryWhileSummarizing = true;
```

Comment:
Here is a problem: we test this value only in lockedSummaryAction(). So previous, if second summary was scheduled while first one was in flight, then once first summary completes, it would trigger another summary.
lockedRefreshSummaryAckAction() does not check it, so if summary was essentially cancelled because of pending lockedRefreshSummaryAckAction() call, then nobody will restart new summary.

I suggest to solve it in a trivial way, not leaking more state across layers / functions. Instead of checking for this.refreshSummaryAckLock !== undefined and bailing out, instead await it. That way, once refresh is over, this code can move forward with summary.